libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1645,13 +1656,27 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		m_context.appendConditionalInvalid();
 
 		if (_operator == Token::Div)
-			m_context << (c_isSigned ? Instruction::SDIV : Instruction::DIV);
+		{
+			if (fixedType)
+			{
+				solUnimplementedAssert(fixedFactor < pow(bigint(2), 256 - fixedType->numBits()), "Not yet implemented - large FixedPointType division");
+				// Division would cancel out factor, this gives the numerator an extra factor so it remains in result
+				m_context << fixedFactor
+						  << Instruction::MUL;
+			}
+			m_context << (isSigned ? Instruction::SDIV : Instruction::DIV);
+		}
 		else
-			m_context << (c_isSigned ? Instruction::SMOD : Instruction::MOD);
+			m_context << (isSigned ? Instruction::SMOD : Instruction::MOD);
 		break;
 	}
 	case Token::Exp:
-		m_context << Instruction::EXP;
+		if (intType)
+			m_context << Instruction::EXP;
+		else
+		{
+			solUnimplemented("Not yet implemented - FixedPointType exponentiation");
```

Comment:
Should this be a regular assert instead because it is disabled in the type system? Also probably swap the if statement to match the others: `if (fixedType)`